guides/testing.md

```diff
@@ -0,0 +1,170 @@
+# Testing Crystal Code
+
+Crystal comes with a full-feature spec library in the [`Spec`module](https://crystal-lang.org/api/latest/Spec.html). It provides the structure for writing executable examples of how your code should behave.
```

Comment:
It's not full-feature. There's no `setup` and `teardown`. That's why I think documenting this right now is not good. Maybe the build-in test library needs to change to have these features. But I doubt that can be done with `describe` and `it`. So maybe we'll have to switch to something like unit test or minitest? I don't know. But that's why I'd prefer not to document this yet.

Reply:
Okay, let's not call it `full-feature` (though that certainly depends).

Reply:
@asterite I strongly disagree btw, there's nothing that the spec library can't do currently which isn't better expressed using helper methods. The spec harness currently is fantastic.